wgpu/src/api/render_pipeline.rs

```diff
@@ -75,20 +76,39 @@ pub struct VertexState<'a> {
     /// Advanced options for when this pipeline is compiled
     ///
     /// This implements `Default`, and for most users can be set to `Default::default()`
+    #[builder(default)]
     pub compilation_options: PipelineCompilationOptions<'a>,
     /// The format of any vertex buffers used with this pipeline.
+    #[builder(default)]
     pub buffers: &'a [VertexBufferLayout<'a>],
 }
 #[cfg(send_sync)]
 static_assertions::assert_impl_all!(VertexState<'_>: Send, Sync);
 
+impl<'a> VertexState<'a> {
+    #[allow(missing_docs)]
+    pub fn from_module(
+        module: &'a ShaderModule,
+    ) -> VertexStateBuilder<
+        'a,
+        (
+            bon::private::Set<&'a ShaderModule>,
+            bon::private::Unset<bon::private::Optional>,
+            bon::private::Unset<bon::private::Optional>,
+            bon::private::Unset<bon::private::Optional>,
+        ),
+    > {
+        Self::builder().module(module)
+    }
+}
```

Comment:
I'm currently considering such syntax for this feature:

```rust
    #[derive(Builder)]
    struct Sut {
        #[builder(pos = start_fn)]
        starter_1: bool,
        
        #[builder(pos = start_fn)]
        starter_2: char,

        named: u32,

        #[builder(pos = finish_fn)]
        finisher_1: &'static str,

        #[builder(pos = finish_fn)]
        finisher_2: &'static str,
    }

    Sut::builder(true, 'c')
        .named(99)
        .build("1", "2");
```
And it's already possible to customize the `builder/build` method names with `#[builder(start_fn/finish_fn = ...)]` at the top-level of the struct

Reply:
Hey @ErichDonGubler, I've just implemented support for the syntax above which allows you to specify with `#[builder(pos = start_fn/finish_fn)]` that the parameter should be passed as a positional parameter of the `start_fn` or `finish_fn`. PR link: https://github.com/elastio/bon/pull/125.

It's also possible to combine this with `#[builder(into)]`, in which case the parameter will be accepted as `impl Into<T>` in the input.

Note that I haven't tested this feature exhaustively yet, but I [tested some simple cases](https://github.com/elastio/bon/blob/1eea830803dabb220ff1988cb9e48d7bd98a0aa6/bon/tests/integration/builder/positional_members.rs#L19-L145). I'll continue testing it and adding documentation for this feature. Hopefully, by the end of this week this will be released in a new minor release.

If you'd like to try this out, you can add a dependency on the dev version of `bon` from my branch like this:

```rust
bon = { git = "https://github.com/elastio/bon", branch = "feat/positional-only-members" }
```

Reply:
I've released this change in a `2.3` version of `bon`. @ErichDonGubler let me know if it helps 🐱 

- [Reddit post](https://www.reddit.com/r/rust/comments/1fgmbo7/media_nextgen_builder_macro_bon_23_release/)
- [Blog post](https://elastio.github.io/bon/blog/bon-builder-v2-3-release)